discovery.md

```diff
@@ -538,6 +539,15 @@ This MUST return an array of zero or more Services. The array MUST contain all
 Services available via this Discovery Enpoint. Any Service previously returned
 to a client that does not appear in this result can be assumed to be deleted.
 
+The collection of services MAY be filtered by supplying an attribute as a query
+parameter.  Doing so will cause only Services with an exact match for that
```

Comment:
I think there needs to be a MUST in here on the DE side so they don't just ignore the filter. But maybe you cover that later on....

Reply:
I think we need to be more explicit about the shape of the query params... eg. its name (and case) related to the attributes? Do we support nested attributes?

Reply:
I'd like to understand a little better the benefit adding these characters brings.  I thought it fairly clear that the names as given earlier in the document (e.g. 249 for `name`) would apply here and I could add something but is "...an attribute `(from the Service attributes above)` as a query..." sufficient to satsify?  Is there better language that you would recommend?

Reply:
A motivational note: I hoped that the phrasing given left space for implementers to provide richer capabilities (e.g. [LIKE semantics](https://docs.microsoft.com/en-us/sql/t-sql/language-elements/like-transact-sql?view=sql-server-ver15)).  To say it differently, I had hoped that exact match would be a dependable minimum bar for supported filters.

Reply:
Even if we think it should be obvious that the url should be `?name=foo` instead of `?Name=foo` I think it would be good to be explicit since parts of URLs are case insensitive.   However, I think also need to be clear on the "exact match" part. Yes it's pedantic, but I think being clear that we also mean case matching too would be good.

As for unknown query parameters, we could just say that the server MUST generate an error if there are unknown query parameters, but that through an out-of-band agreement servers MAY extend the list of query parameters.

Reply:
Two changes to call out:

> Doing so MUST cause only Services with at least an exact value match for those filters to be
included in the result.

`at least` was inserted.  This allows for fuzzy matching and other above-and-beyond capabilities (the intention is to set a minimum bar rather than limit implementations to the minimum bar)

Later, after requiring `name` filtering support and the explicit rejection of unsupported filters...

>   Endpoints MAY add additional query parameters or filter processing to improve upon this minimum provision.

Calling these out since it risks going slightly beyond the approved changes but I think captures the intent more thoroughly.  Clearly this can be altered by follow on PRs.